REPL completions: Enter import mode only when cursor beyond "import"

xal-0 opened the pull request:
Adds a position argument to `get_import_mode`.

Fixes #55842

xal-0 committed:
REPL completions: Enter import mode only when cursor beyond "import"

Fixes #55842

vtjnash commented on stdlib/REPL/src/REPLCompletions.jl:
```diff
@@ -939,13 +939,16 @@ const superscripts = Dict(k[3]=>v[1] for (k,v) in latex_symbols if startswith(k,
     # ^\s*(?:@\w+\s*(?:\(\s*)?)?
 
+    # Do not enter import mode unless cursor beyond import keyword
+    beyond_kw(m) = pos >= m.offsets[1] + length(m[1])
```
```suggestion
    beyond_kw(m) = pos >= m.offsets[1] + sizeof(m[1])
```

Commit:
Update stdlib/REPL/src/REPLCompletions.jl

xal-0 committed:
REPL completion: make import mode more forgiving

Don't require a using/import to be on its own line to enter import mode; instead
look for partial import expressions and require the cursor be in the range where
modules/imported names would go.

Fixes #56389 (including the fix for #55842)

xal-0 commented:
Closing in favour of #57767
